### PR TITLE
Savestates: Fixes Maxima

### PR DESCRIPTION
**Pull request by elad335**
* Fix a deadlock regarding VM locks with emulation stopping and savestate loading.
* Fixup emulation stopping.
* Fix SPU LLVM sinking stores beyond a saveable code point.
* Make saving state impossible while a cellSaveData operation is active.
* Save pending RSX flip, fixes savestate freeze on load with Tales Of Xillia 2, or I guess, every game that needs PS3 Native frame rate mode.

I can't anymore find bugs in games with savestates anymore, if you do, please report.

### CODE REVIEW

File: rpcs3/Emu/Cell/Modules/cellGem.cpp [hunk 1]
```diff
@@ -1620,7 +1620,7 @@ error_code cellGemGetRumble(u32 gem_num, vm::ptr<u8> rumble)
 
 error_code cellGemGetState(u32 gem_num, u32 flag, u64 time_parameter, vm::ptr<CellGemState> gem_state)
 {
-	cellGem.warning("cellGemGetState(gem_num=%d, flag=0x%x, time=0x%llx, gem_state=*0x%x)", gem_num, flag, time_parameter, gem_state);
+	cellGem.trace("cellGemGetState(gem_num=%d, flag=0x%x, time=0x%llx, gem_state=*0x%x)", gem_num, flag, time_parameter, gem_state);
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/cellGem.cpp:1623]: don't trace this yet

File: rpcs3/Emu/System.h [hunk 2]
```diff
@@ -401,6 +401,25 @@ class Emulator final
 	static void SaveSettings(const std::string& settings, const std::string& title_id);
 };
 
+struct hle_locks_t
```
Comment (Megamouse) [inline rpcs3/Emu/System.h:404]: For the love of god, stop dumping everything into System

Comment (elad335) [reply rpcs3/Emu/System.h]: Fixed,

Comment (Megamouse) [reply rpcs3/Emu/System.h]: thanks

**Pull request merged**